src/strawberry_sqlalchemy_mapper/relay.py

```diff
@@ -0,0 +1,427 @@
+    )
+
+    @classmethod
+    def resolve_connection(
+        cls,
+        nodes: Union[Query, StrawberrySQLAlchemyAsyncQuery],  # type: ignore[override]
+        *,
+        info: Info,
+        before: Optional[str] = None,
+        after: Optional[str] = None,
+        first: Optional[int] = None,
+        last: Optional[int] = None,
+        **kwargs: Any,
+    ) -> AwaitableOrValue[Self]:
+        from .field import StrawberrySQLAlchemyAsyncQuery, connection_session
+
+        if first and last:
+            raise ValueError("Cannot provide both `first` and `last`")
+        elif first and before:
+            raise ValueError("`first` cannot be provided with `before`")
+        elif last and after:
+            raise ValueError("`last` cannot be provided with `after`")
+
+        max_results = info.schema.config.relay_max_results
+        per_page = first or last or max_results
+        if per_page > max_results:
+            raise ValueError(f"Argument 'last' cannot be higher than {max_results}.")
+
+        session = connection_session.get()
+        assert session is not None
+
+        def resolve_connection(page: sqlakeyset.Page):
+            type_def = get_object_definition(cls)
+            assert type_def
+            field_def = type_def.get_field("edges")
+            assert field_def
+
+            field = field_def.resolve_type(type_definition=type_def)
+            while isinstance(field, StrawberryContainer):
+                field = field.of_type
+
+            edge_class = cast(Edge[NodeType], field)
+
+            return cls(
+                page_info=relay.PageInfo(
+                    has_next_page=page.paging.has_next,
+                    has_previous_page=page.paging.has_previous,
+                    start_cursor=page.paging.get_bookmark_at(0) if page else None,
+                    end_cursor=page.paging.get_bookmark_at(-1) if page else None,
+                ),
+                edges=[
+                    edge_class.resolve_edge(n, cursor=page.paging.get_bookmark_at(i))
+                    for i, n in enumerate(page)
+                ],
+            )
+
+        def resolve_nodes(s: Session, nodes=nodes):
+            if isinstance(nodes, StrawberrySQLAlchemyAsyncQuery):
+                nodes = nodes.query(s)
+
+            return resolve_connection(
+                sqlakeyset.get_page(
+                    nodes,
+                    before=(
+                        sqlakeyset.unserialize_bookmark(before).place
+                        if before
+                        else None
+                    ),
+                    after=(
+                        sqlakeyset.unserialize_bookmark(after).place if after else None
+                    ),
+                    per_page=per_page,
+                )
+            )
+
+        if isinstance(session, AsyncSession):
+
+            async def resolve_async(nodes=nodes):
+                return await session.run_sync(lambda s: resolve_nodes(s))
```

Comment:
I had to use the `sqlakeyset.get_page` API instead of the `sqlakeyset.select_page` because of the way this is implemented.

sqlakeyset doesn't have a `get_page` async counterpart, only a `select_page`

TBH I don't know much of the sqlalchemy internals and neither of `sqlakeyset`, so open to try any suggestions here =P